### PR TITLE
♻️ adds support for bookend config v1.0 in amp story v0.1.

### PR DESCRIPTION
**Pull request by Enriqe**
Since in some cases publishers might update their JSON config to use the 1.0 spec, they might not yet have the markup ready to comply with the amp-story 1.0 spec. We want to support this case to enable an easier transition.

### CODE REVIEW

File: extensions/amp-story/0.1/related-articles.js [hunk 1]
```diff
@@ -91,3 +91,25 @@ export function relatedArticlesFromJson(opt_articleSetsResponse) {
...
+ * @param {!JsonObject=} bookendComponents
+ * @return {!Array<!RelatedArticleSetDef>}
+ */
+export function parseArticlesToClassicApi(bookendComponents) {
+  const articleSet = {};
+  articleSet.articles = [];
+
+  bookendComponents.forEach(component => {
+    if (component.type != 'small') {
+      user().warn(component.type + ' is not supported in amp-story-0.1, ' +
```
Comment (newmuis) [inline extensions/amp-story/0.1/related-articles.js]: This is not necessarily true; they could put `component.type` as something completely unsupported.  I'm fine with hardcoding the list of stuff from 1.0 here.

Comment (Enriqe) [reply extensions/amp-story/0.1/related-articles.js]: Done.

File: extensions/amp-story/0.1/related-articles.js [hunk 2]
```diff
@@ -91,3 +91,30 @@ export function relatedArticlesFromJson(opt_articleSetsResponse) {
...
     }));
 }
+
+/**
+ * @param {!JsonObject=} bookendComponents
+ * @return {!Array<!RelatedArticleSetDef>}
+ */
+export function parseArticlesToClassicApi(bookendComponents) {
+  const articleSet = {};
+  const newComponents =
```
Comment (newmuis) [inline extensions/amp-story/0.1/related-articles.js]: nit: move to top level and rename `NEW_COMPONENTS`

**Comment by Enriqe** [comment]
ready to merge @newmuis

**Pull request merged**